relentless/simulate/hoomd.py

```diff
@@ -175,27 +176,33 @@ def extract_box_params(self, sim):
         Raises
         ------
         ValueError
-            If the volume is not set.
+            If the extent is not set.
         TypeError
-            If the volume does not derive from :class:`~relentless.volume.TriclinicBox`.
+            If the extent does not derive from :class:`~relentless.extent.TriclinicBox` or :class:`~relentless.extent.ObliqueArea`.
 
         """
         # cast simulation box in HOOMD parameters
         V = sim.ensemble.V
         if V is None:
-            raise ValueError('Box volume must be set.')
-        elif not isinstance(V, TriclinicBox):
-            raise TypeError('HOOMD boxes must be derived from TriclinicBox')
-
-        Lx = V.a[0]
-        Ly = V.b[1]
-        Lz = V.c[2]
-        xy = V.b[0]/Ly
-        xz = V.c[0]/Lz
-        yz = V.c[1]/Lz
-
-        return numpy.array([Lx,Ly,Lz,xy,xz,yz])
-
+            raise ValueError('Box extent must be set.')
+        elif not isinstance(V, (TriclinicBox, ObliqueArea)):
+            raise TypeError('HOOMD boxes must be derived from TriclinicBox or ObliqueArea')
+        if isinstance(V, TriclinicBox):
+            Lx = V.a[0]
+            Ly = V.b[1]
+            Lz = V.c[2]
+            xy = V.b[0]/Ly
+            xz = V.c[0]/Lz
+            yz = V.c[1]/Lz
+            return numpy.array([Lx,Ly,Lz,xy,xz,yz])
+        elif isinstance(V, ObliqueArea):
+            Lx = V.a[0]
+            Ly = V.b[1]
+            Lz = 1
+            xy = V.b[0]/Ly
+            xz = 0
+            yz = 0
+            return numpy.array([Lx,Ly,Lz,xy,xz,yz])
```

Comment:
Adding in the dimensionality breaks quite a few tests downstream that I couldn't resolve. The simulation box doesn't return a dimensionality which is why I just hard-coded it based on class. 

Does this need to be in this function, or can it be stored in sim[self].dim as discussed below?

Reply:
I think it would be convenient to return the dimensionality here... let's discuss on Slack if you can document the errors you got when you made this change.

Reply:
I was able to get it resolved. I had forgotten to unpack the dimension from the gsd, so it triggered a ton of mismatched array errors. All resolved now!